### ISSUE TITLE
Out of memory when trainer.save_checkpoint("example.ckpt")

### ISSUE THREAD

**Issue by VirajBagal** [PyTorchLightning/pytorch-lightning#3582]
The sbatch session crashes and I get the following error when I include trainer.save_checkpoint("example.ckpt") in my code.

> /var/spool/slurmd/job220424/slurm_script: line 15: 39865 Killed                  python roco_train_mlm_lightning.py --run_name debug --precision 16 --mlm_prob 0.15
> slurmstepd: error: Detected 3 oom-kill event(s) in step 220424.batch cgroup. Some of your processes may have been killed by the cgroup out-of-memory handler.

Same thing happens when I run the code in notebook on the remoteserver. The kernel dies. Please help. Thank you.

**Comment by ananyahjha93** [comment]
@VirajBagal which version of lightning are you using? Currently I am not able to reproduce this with 0.9.0.

**Comment by VirajBagal** [comment]
@ananyahjha93  Let me give some details.

```
    def validation_step(self, batch, batch_idx):

        loss, acc = self.shared_step(batch, batch_idx)
        result = pl.EvalResult(loss)

        container = {'val_loss': loss, 'val_acc': acc}        
        result.log_dict(container, on_step = True, on_epoch = True, prog_bar = True, logger = True)

        return result
```
Now, I want to save checkpoints/modelweights at the end of every validation epoch if my 'total val loss' is smaller than previous 'best val loss'. How do I do that?  The output is following:

```

```
Epoch 0:  87%|████████▋ | 13/15 [00:05<00:00,  2.56it/s, loss=2.698, v_num=08zk, step_train_loss=1.17, step_train_acc=0]
Epoch 0:  93%|█████████▎| 14/15 [00:05<00:00,  2.66it/s, loss=2.698, v_num=08zk, step_train_loss=1.17, step_train_acc=0]
Epoch 0: 100%|██████████| 15/15 [00:05<00:00,  2.58it/s, loss=2.698, v_num=08zk, step_train_loss=1.17, step_train_acc=0, step_val_loss=1.08, step_val_acc=0, epoch_val_loss=1.11, epoch_val_acc=0]
```
```

I tried manually ModelCheckpoint as well:

```
checkpoint_callback = ModelCheckpoint(
    filepath=os.getcwd(),
    save_top_k=1,
    verbose=True,
    monitor='epoch_val_loss',
    mode='min',
    prefix=''
)
```

But it didnt help. I get the following error: 

> /home/viraj.bagal/anaconda3/envs/medvqa/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py:37: UserWarning: 
>                     When using EvalResult(early_stop_on=X) or TrainResult(early_stop_on=X) the
>                     'monitor' key of ModelCheckpoint has no effect.
>                     Remove ModelCheckpoint(monitor='epoch_val_loss) to fix')
>                 
>   warnings.warn(*args, **kwargs)
> /home/viraj.bagal/anaconda3/envs/medvqa/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py:37: RuntimeWarning: The metric you returned None must be a `torch.Tensor` instance, checkpoint not saved HINT: what is the value of epoch_val_loss in validation_epoch_end()?
>   warnings.warn(*args, **kwargs)
> /home/viraj.bagal/anaconda3/envs/medvqa/lib/python3.7/site-packages/pytorch_lightning/utilities/distributed.py:37: RuntimeWarning: Can save best model only with epoch_val_loss available, skipping.
>   warnings.warn(*args, **kwargs)

**Comment by VirajBagal** [comment]
Sorry. Its my mistake. This solved the issue.
`python
result = pl.EvalResult(checkpoint_on = loss)`
